### DISCUSSION TITLE
Trying to configure a coursier behind a proxy repository

### DISCUSSION THREAD

**Discussion by DamianReeves** [coursier/coursier#2404] [Q&A]
My company has a proxy repository which all artifact access is fronted by. 
As such in sbt, we use the configuration details mentioned here: https://www.scala-sbt.org/1.x/docs/Proxy-Repositories.html#sbt+Configuration in order to configure sbt to correctly resolve artifacts.

The docs referenced speak to a block like this:

```
[repositories]
  local
  my-ivy-proxy-releases: http://repo.company.com/ivy-releases/, [organization]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)[revision]/[type]s/[artifact](-[classifier]).[ext]
  my-maven-proxy-releases: http://repo.company.com/maven-releases/
```

This works great for sbt, but I am trying to accomplish the same in the coursier CLI (as well as in mill, which I know is outside the scope of this repo).
What would be the raw coursier configuration way to accomplish what sbt does? I'm interested both in how to do this with config files, i.e. using `coursier.mirrors`, using environment variables and finally using command line options.

What's of particular note is how to configure the ivy pattern in coursier as shown in sbt.

**Comment by Quafadas** [comment]
Have you come across this page? IT's documented for metals, but I found it to be quite a general solution... 

https://scalameta.org/metals/docs/troubleshooting/proxy/
